### DISCUSSION TITLE
New behavior when checking stuff in and out

### DISCUSSION THREAD

**Discussion by floschoepfer** [snipe/snipe-it#13474] [Ideas]
## Current Behavior
Currently, when you check stuff (Assets, Accessories, Licences, etc.) in or out, you'll always be redirected to the "main" page of the item type.
For example: I'm on the page of the Asset category "Laptops" and now I'm checking an Asset in (it doesn't matter if I go to the Asset page or just do it with the button on the table). After the check-in, I always get redirected to the main Assets page and not to the page that I was before (in my example, the page of the category "Laptops").

## How I would like it
I would like that when I check stuff out (or in), that I get redirected to the page I was on before and not always to the main page of the item type.

### Current Behavior
![16-08-_2023_07-57-08](https://github.com/snipe/snipe-it/assets/112496896/4da3022c-d27c-4cf3-9fca-73f5c9fe2354)
After the checkout, I get redirected to the main Assets page

### How I would like it
![16-08-_2023_07-55-23](https://github.com/snipe/snipe-it/assets/112496896/876c921b-5c1b-4ac0-99f5-1baee954880f)
After the checkout, I should get redirected to the page I was on before (in this example to the page of this Asset)

## Notes
Maybe you can even give us an option in the settings where we can define what will happen after a check in or out (redirected to the main page or redirected to the page I was on before)?

@snipe what are your thoughts about this?

**Comment by floschoepfer** [comment]
@snipe did you get the time to look into this? Or should I create a Feature Request out of it?

**Reply by snipe** [reply]
A feature request already exists on our internal tracker.
